### PR TITLE
[video_player] Relands #6456: Uses `SurfaceProducer`, this time with `setCallback` for suspend/resume lifecycles.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/139702.

/cc @jonahwilliams @chinmaygarde @johnmccutchan.

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
Does phone rotation impact the exo video? Are there ways to configure video player to play a video from a hardware device (camera?) that would require rotation?

**Comment by matanlurey** [comment]
As mentioned offline, I'll need to merge this into a preview branch (maybe just `impeller`) until the next stable.

@stuartmorgan @camsim99 Would you prefer I either:

1. Merge all of the changes that _can_ land right now, i.e. break this into two PRs.
2. Not bother, and instead merge against a new `impeller` branch.

**Comment by camsim99** [comment]
I have no strong preference but breaking into two PRs may be easier for
packages contributors to follow, as I'd assume we'd otherwise need to
document the impeller branch somewhere for them to be aware.

**Comment by xxoo** [comment]
@matanlurey
I think saving and restoring the player state is not always acceptable. Please consider about [this proposal](https://github.com/flutter/flutter/issues/152839#issue-2447195040). Thanks

**Comment by matanlurey** [comment]
Ping @camsim99 @gmackall I think this is ready to land (modulo repo checks), could you do a final LGTM?

**Review by camsim99 — APPROVED** [review]
At a high level this LGTM!! There is a [synthetic accessor lint error](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8738879054698201169/+/u/Run_package_tests/lint/stdout#:~:text=snapshot%0A%20%20%0A%20%20Lint%20found%2010%20errors%2C%200%20warnings.%20First%20failure%3A-,/b/s/w/ir/x/w/packages/packages/video_player/video_player_android/android,method%20createVideoPlayer%20of%20class%20VideoPlayer%20requires%20synthetic%20accessor%20%5BSyntheticAccessor%5D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20setExoPlayer(createVideoPlayer())%3B,-~~~~~~~~~~~~~~~~~%0A%20%20%0A%20%20The%20full%20lint%20text%20report%20is%20located%20at%3A%0A%20%20%20%20/b) though

**Review by gmackall — APPROVED** [review]

File: packages/video_player/video_player_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,172 +1,176 @@
+## 2.7.0
+
+- Re-adds [support for Impeller](https://docs.flutter.dev/release/breaking-changes/android-surface-plugins).
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply packages/video_player/video_player_android/CHANGELOG.md:3]: Nope, happy to change back.

**Comment by Gustl22** [comment]
Relanded in #7498

**Pull request merged**